src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,187 @@
+  const initialCenturyParams = initialSearchParams.getAll('century');
+
+  const initialSortQuery = initialSearchParams.get('sort') || SortQuery.INITIAL;
+  const initialOrderQuery = initialSearchParams.get('order') || SortOrder.ASC;
+
+  const [searchQuery, setSearchQuery] = useState(initialSearchQuery);
+  const [sex, setSex] = useState(initialSexParam);
+  const [centuries, setCenturies] = useState(initialCenturyParams);
+
+  const [sortQuery, setSortQuery] = useState<SortQuery>(
+    initialSortQuery as SortQuery,
+  );
+  const [sortOrder, setSortOrder] = useState<SortOrder>(
+    initialOrderQuery as SortOrder,
+  );
+  const [sortedList, setSortedList] = useState<Person[]>([]);
+
+  const initialList = useMemo(() => {
+    if (filteredList) {
+      return [...filteredList];
+    }
+
+    return [];
+  }, [filteredList]);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    fetchPeopleData()
+      .then((data) => {
+        setPeopleData(data);
+      })
+      .catch(() => {
+        setIsLoading(false);
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (!peopleData) {
+      return;
+    }
+
+    const filteredData = peopleData.filter((person) => {
+      const nameMatch
+        = initialSearchQuery === ''
+        || person.name.toLowerCase().includes(initialSearchQuery)
+        || person.motherName?.toLowerCase().includes(initialSearchQuery)
+        || person.fatherName?.toLowerCase().includes(initialSearchQuery);
+
+      const sexMatch = initialSexParam === '' || person.sex === initialSexParam;
+
+      const centuryMatch
+        = initialCenturyParams.length === 0
+        || initialCenturyParams.includes(calcCentury(person.born).toString())
+        || initialCenturyParams.includes(calcCentury(person.died).toString());
+
+      return nameMatch && sexMatch && centuryMatch;
+    });
+
+    setFilteredList(filteredData);
+  }, [searchQuery, peopleData, sex, centuries.length]);
+
+  useEffect(() => {
+    const newSortedList = [...filteredList || []];
+
+    if (sortQuery === INITIAL && initialList) {
+      setSortedList([...initialList]);
+    } else {
+      newSortedList.sort((person1, person2) => {
+        switch (sortQuery) {
+          case NAME:
+          case SEX: {
+            if (sortOrder === ASC) {
+              return person1[sortQuery].localeCompare(person2[sortQuery]);
+            }
+
+            return person2[sortQuery].localeCompare(person1[sortQuery]);
+          }
+
+          case BORN:
+          case DIED: {
+            if (sortOrder === ASC) {
+              return person1[sortQuery] - person2[sortQuery];
+            }
+
+            return person2[sortQuery] - person1[sortQuery];
+          }
+
+          default:
+            return 0;
+        }
+      });
+
+      setSortedList(newSortedList);
+    }
+  }, [sortQuery, sortOrder, initialList]);
```

Comment:
The dependencies array in the useEffect hook is missing the 'filteredList' dependency.